kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -200,6 +202,37 @@ showLeafs = do
     showPair :: (NodeStates, [Graph.Node]) -> String
     showPair (ns, xs) = show ns <> ": " <> show xs
 
+showPrecBranch
+    :: MetaOrObject level
+    => Maybe Int
+    -> StateT (ReplState level) Simplifier ()
+showPrecBranch mnode = do
+    Strategy.ExecutionGraph { graph } <- Lens.use lensGraph
+    node <- Lens.use lensNode
+    let node' = maybe node id mnode
+    if node' `elem` Graph.nodes graph
+        then do
+            putStrLn' $ show (findBranch graph [node'])
+        else putStrLn' "Invalid node!"
+  where
+    findBranch gph xs
```

Comment:
I think this could be rewritten using `loop` (https://hackage.haskell.org/package/extra-1.6.14/docs/Control-Monad-Extra.html#v:loop) where `a` is `Graph.Node` and `b` is `[Graph.Node]`:

You should always have `outdeg` exactly 1 until you find a branch, no? So it should be enough to have only `Left Graph.Node` until you are done, when you return `Right [...]`.